cli/cmd/plugin/unmanaged-cluster/log/log.go

```diff
@@ -180,66 +179,89 @@ func (l *CMDLogger) Warn(message string) {
 		return
 	}
 
+	// default to hi intensity yellow when color is not set
+	if l.logColor == color.Reset {
+		l.logColor = color.FgHiYellow
+	}
+
 	message = processStyle(l, message)
 	fmt.Print(message)
+	l.logColor = color.Reset
 }
 
 func (l *CMDLogger) Warnf(message string, args ...interface{}) {
 	if l.logLevel > l.level {
 		return
 	}
 
+	// default to hi intensity yellow when color is not set
+	if l.logColor == color.Reset {
+		l.logColor = color.FgHiYellow
+	}
+
 	message = processStyle(l, message)
 	fmt.Fprintf(l.output, message, args...)
+	l.logColor = color.Reset
 }
 
 func (l *CMDLogger) Error(message string) {
 	if l.logLevel > l.level {
 		return
 	}
 
-	// default to red when color is not set
-	if l.color == "" {
-		l.color = ColorRed
-		message = processStyle(l, message)
-	} else {
-		message = processStyle(l, message)
+	// default to hi intensity red when color is not set
+	if l.logColor == color.Reset {
+		l.logColor = color.FgHiRed
 	}
+
+	message = processStyle(l, message)
 	fmt.Print(message)
+	l.logColor = color.Reset
 }
 
 func (l *CMDLogger) Errorf(message string, args ...interface{}) {
 	if l.logLevel > l.level {
 		return
 	}
 
-	// default to red when color is not set
-	if l.color == "" {
-		l.color = ColorRed
-		message = processStyle(l, message)
-	} else {
-		message = processStyle(l, message)
+	// default to hi intensity red when color is not set
+	if l.logColor == color.Reset {
+		l.logColor = color.FgHiRed
 	}
+
 	message = processStyle(l, message)
 	fmt.Fprintf(l.output, message, args...)
+	l.logColor = color.Reset
 }
 
 func (l *CMDLogger) Info(message string) {
 	if l.logLevel > l.level {
 		return
 	}
 
+	// default to faint when color is not set
```

Comment:
I love the progression now of how things go from "light" to "faint" -- however, I'm not too crazy about this default.

I wouldn't expect defaults or usage of ColorNone to use `faint`, which **is** a color.

For example:

```
log.Style(0, logger.ColorNone).Infof("View available packages:\n")
```

Produces faint, which is surprising to me.

Is there anyway to invert this behavior such that you need to specify faint, rather than it being the default?

Reply:
> `log.Style(0, logger.ColorNone).Infof("View available packages:\n")`
> Produces faint, which is surprising to me.

Ah - that is surprising! I hadn't considered the `log.Style(...).{Method}` workflow when adding defaults.

Yes, I think the alternative is to invert the behavior, otherwise, we'll be overriding people's color choice. Good catch!